### PR TITLE
Keep track of BRIGHT in final coadd mask, keep ormask

### PR DESCRIPTION
**Pull request by esheldon**
keep track of BRIGHT in the coadd_exp.mask
    
    This coadding mask is currently only being set for EDGE and if it is set
    it is considered a bug and raises an exception.  Also add in BRIGHT and
    SAT anywhere BRIGHT was set in the warps.  The SAT will prevent
    detection in those areas
    
    Updated interp flags to include BRIGHT as it is now required to be added
    to the mask plane by the caller.  No longer setting SAT for BRIGHT
    at that stage
    
    added verification that EDGE and NO_DATA are not set in the warps
    
    added verification that EDGE is not in the coadd
    
    corrected some typos

### CODE REVIEW

**Comment by esheldon** [comment]
I'm currently setting the observation ormask to the exposure mask, and leaving bmask as zeros

I don't think this is actually used downstream, only the exp.mask is used, but might want to be consistent with DES here.

@beckermr what do you recommend?

**Comment by beckermr** [comment]
My recommendation is the following:

1. Compute an OR of all of the flags from the SE input images and put this in the ormask.
2. For the bit mask of the coadd, send in a mask that has flags for interpolated pixels and bright pixels.

**Comment by esheldon** [comment]
Then the mask in the exposure would not map to either the obs.bmask or obs.ormask which could be confusing

**Comment by beckermr** [comment]
Indeed it could be, but I think these masks are the most sensible. The OR mask has data about the inputs to the coadd. The bmask has data about how the coadd was processed in each pixel.

**Comment by esheldon** [comment]
now I'm not sure why we want 2 separate from 1 since it is a subset

**Comment by beckermr** [comment]
That's true. We could store them in the same set of bits.

**Comment by esheldon** [comment]
How about this:

Do 1. as you said for ormask, but make bmask a copy of the mask in the exposure

**Comment by beckermr** [comment]
Yep. We should keep ormask flags for which pixels got inteprolated too.

**Comment by esheldon** [comment]
Do you mean define a new mask plane to propagate?

**Comment by esheldon** [comment]
We could re-use INTERP

but then we would not be able to distinguish between things we interpolated and the stack interpolated, which may be useful especially since with real images both may be happening

**Comment by esheldon** [comment]
But this would in any case be redundant with the ormask, which is propagating whatever we would have labeled as interpolated

File: descwl_coadd/coadd.py [hunk 1]
```diff
@@ -443,35 +495,30 @@ def make_stacker(mask, coadd_dims):
...
 
-    # TODO when we make the BRIGHT plane, we will have BRIGHT at 0.0 here but
-    # not so for others (e.g. SAT should be 0.1 or whatever)
-    # TODO make this part of a configuration
+    # the mask here is going to be just EDGE and we always
+    # want to watch for it. it is a bug if EDGE is included
+    # for regular images (not psf)
+    mask_prop_thresh = {}
+    for flagname in FLAGS2CHECK_FOR_COADD:
+        mask_prop_thresh[flagname] = 0.001
```
Comment (beckermr) [inline descwl_coadd/coadd.py:520]: Why the magic number 0.001 vs 0?

Comment (esheldon) [reply descwl_coadd/coadd.py:520]: depends on if the test is `>` or `>=`

**Review by beckermr — CHANGES_REQUESTED** [review]
One question on a magic number. 

Also, we should download the CATSIM stuff and the bright stars and run those tests in the CI. I don't think we are doing this currently.

File: descwl_coadd/coadd.py (hunk 1 above)
Comment (esheldon) [reply descwl_coadd/coadd.py:520]: just checked, the test is `>` so we need some small finite number, 0.001 seems ok

Comment (esheldon) [reply descwl_coadd/coadd.py:520]: ugh sorry had it backwards, that means zero is fine

Comment (beckermr) [reply descwl_coadd/coadd.py]: I am confused. Can you explain in more detail?

Comment (beckermr) [reply descwl_coadd/coadd.py]: Ahhh nvm.

Comment (esheldon) [reply descwl_coadd/coadd.py]: no problem, was in a hurry as I need to get on the road

**Comment by esheldon** [comment]
is this ok to merge?

**Comment by beckermr** [comment]
Did you want to add the CATSIM download the CI job? Happy to do that in another PR too.

**Comment by esheldon** [comment]
thanks, let's do it in a different PR

**Pull request merged**